lib/p2p/Peer.ts

```diff
@@ -474,34 +498,34 @@ class Peer extends EventEmitter {
     return packet;
   }
 
-  private handleHello = (packet: HelloPacket): void => {
-    this.handshakeState = { ...this.handshakeState, ...packet.body };
+  private handleHello = (packet: packets.HelloPacket): void => {
+    const entry = this.responseMap.get(PacketType.HELLO);
 
-    if (this.responseMap.has(packet.type)) {
-      this.fulfillResponseEntry(packet.type, packet);
+    if (!entry) {
+      this.logger.debug(`Peer (${this.id}) sent an unsolicited Hello packet`);
+      // TODO: penalize
+    } else {
+      this.responseMap.delete(PacketType.HELLO);
+      entry.resolve(packet);
```

Comment:
We have a bit of code duplication here, but that's fine if you want to close `fulfillResponseEntry` to the usual use-case.

Reply:
Yeah it was intentional, I think it's different enough and it's clearly a special case.